tests/observability/metrics/utils.py

```diff
@@ -1342,3 +1358,119 @@ def get_vmi_guest_os_kernel_release_info_metric_from_vm(
         "vmi_pod": vm.vmi.virt_launcher_pod.name,
     }
+
+
+def wait_for_file_system_metric_mountpoints_existence(
+    prometheus: Prometheus,
+    capacity_or_used: str,
+    vm: VirtualMachineForTests,
+    dfs_info: dict[str, dict[str, str]],
+) -> None:
+    samples = TimeoutSampler(
+        wait_timeout=TIMEOUT_2MIN,
+        sleep=TIMEOUT_15SEC,
+        func=metric_result_output_dict_by_mountpoint,
+        prometheus=prometheus,
+        capacity_or_used=capacity_or_used,
+        vm_name=vm.name,
+    )
+    mount_points_with_value_zero = None
+    try:
+        for sample in samples:
+            if sample:
+                if [mount_point for mount_point in dfs_info if not sample.get(mount_point)]:
+                    continue
+                mount_points_with_value_zero = {
+                    mount_point: float(sample[mount_point]) for mount_point in sample if int(sample[mount_point]) == 0
+                }
+                if not mount_points_with_value_zero:
+                    return
+    except TimeoutExpiredError:
+        LOGGER.error(f"There is at least one mount point with value zero: {mount_points_with_value_zero}")
+        raise
+
+
+def get_pvc_size_bytes(vm: VirtualMachineForTests) -> str:
+    return str(
+        int(
+            bitmath.parse_string_unsafe(
+                PersistentVolumeClaim(
+                    name=vm.instance.spec.dataVolumeTemplates[0].metadata.name,
+                    namespace=vm.namespace,
+                ).instance.spec.resources.requests.storage
+            ).Byte.bytes
+        )
+    )
+
+
+def get_vm_virt_launcher_pod_requested_memory(vm: VirtualMachineForTests) -> int:
+    return int(
+        bitmath.parse_string_unsafe(
+            vm.vmi.virt_launcher_pod.instance.spec.containers[0].resources.requests.memory
+        ).bytes
+    )
+
+
+def virt_launcher_pod_metrics_resource_exists(vm_for_test: VirtualMachineForTests) -> None:
+    vl_name = vm_for_test.vmi.virt_launcher_pod.name
+    samples = TimeoutSampler(
+        wait_timeout=TIMEOUT_1MIN,
+        sleep=TIMEOUT_15SEC,
+        func=lambda: PodMetrics(name=vl_name, namespace=vm_for_test.namespace).exists,
+    )
+    try:
+        for sample in samples:
+            if sample:
+                LOGGER.info(f"PodMetric resource for {vl_name} exists.")
+                return
+    except TimeoutExpiredError:
+        LOGGER.error(f"Resource PodMetrics for pod {vl_name} not found")
+        raise
+
+
+def get_vm_memory_working_set_bytes(vm: VirtualMachineForTests) -> int:
+    virt_launcher_pod_metrics_resource_exists(vm_for_test=vm)
+    match = re.search(
+        r"\b(\d+Mi)\b",
+        run_command(
+            command=shlex.split(f"oc adm top pod {vm.vmi.virt_launcher_pod.name} -n {vm.namespace}"),
+            check=False,
+        )[1],
+    )
+    assert match, "vm memory working set not found"
+    return int(bitmath.parse_string_unsafe(match.group(1)).bytes)
```

Comment:
can be combined and re-written using https://github.com/RedHatQE/timeout-sampler/blob/main/timeout_sampler/__init__.py#L256

Reply:
i think it is ok to keep these 2 separate  - makes it clear what fails.